File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionContainsTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionContainsTest extends BaseTest {
    @Test
    void object() {
        assertTransformation(fromJson("""
[0, [], "a"]"""), fromJson("""
{
  "$$contains": "$", "that": "a"
}
"""), true);
        // with transformation
        assertTransformation("a", fromJson("""
{
  "$$contains": ["b","$"], "that": "a"
}
"""), true);

        assertTransformation(fromJson("""
[0, [], "a"]"""), fromJson("""
{
  "$$contains": "$", "that": "b"
}
"""), false);
    }

    @Test
    void inline() {
        assertTransformation(fromJson("[0, [], \"a\"]"), "$$contains(a):$", true);
        assertTransformation(fromJson("[0, [], \"a\"]"), "$$contains(b):$", false);
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionContains.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionContains", () => {
  test("object", async () => {
    await assertTransformation(
      [0, [], "a"],
      {
        $$contains: "$",
        that: "a",
      },
      true,
    );
    // with transformation
    await assertTransformation(
      "a",
      {
        $$contains: ["b", "$"],
        that: "a",
      },
      true,
    );

    await assertTransformation(
      [0, [], "a"],
      {
        $$contains: "$",
        that: "b",
      },
      false,
    );
  });

  test("inline", async () => {
    await assertTransformation([0, [], "a"], "$$contains(a):$", true);
    await assertTransformation([0, [], "a"], "$$contains(b):$", false);
  });
});
